v03_pipeline/lib/annotations/shared.py

```diff
@@ -0,0 +1,145 @@
+    'splice_region_variant',
+    'incomplete_terminal_codon_variant',
+    'start_retained_variant',
+    'stop_retained_variant',
+    'synonymous_variant',
+    'coding_sequence_variant',
+    'mature_miRNA_variant',
+    '5_prime_UTR_variant',
+    '3_prime_UTR_variant',
+    'non_coding_transcript_exon_variant',
+    'non_coding_exon_variant',  # deprecated
+    'intron_variant',
+    'NMD_transcript_variant',
+    'non_coding_transcript_variant',
+    'nc_transcript_variant',  # deprecated
+    'upstream_gene_variant',
+    'downstream_gene_variant',
+    'TFBS_ablation',
+    'TFBS_amplification',
+    'TF_binding_site_variant',
+    'regulatory_region_ablation',
+    'regulatory_region_amplification',
+    'feature_elongation',
+    'regulatory_region_variant',
+    'feature_truncation',
+    'intergenic_variant',
+]
+CONSEQUENCE_TERMS_LOOKUP = hl.dict(hl.enumerate(CONSEQUENCE_TERMS, index_first=False))
+
+OMIT_CONSEQUENCE_TERMS = hl.set(
+    [
+        'upstream_gene_variant',
+        'downstream_gene_variant',
+    ],
+)
+
+SELECTED_ANNOTATIONS = [
+    'amino_acids',
+    'biotype',
+    'canonical',
+    'codons',
+    'gene_id',
+    'hgvsc',
+    'hgvsp',
+    'lof',
+    'lof_filter',
+    'lof_flags',
+    'lof_info',
+    'transcript_id',
+]
+
+
+def sample_id(mt: hl.MatrixTable, **_: Any) -> hl.Expression:
+    return mt.s
+
+
+def rg37_locus(
+    ht: hl.Table,
+    reference_genome: ReferenceGenome,
+    liftover_ref_path: str,
+    **_: Any,
+) -> hl.Expression | None:
+    if reference_genome == ReferenceGenome.GRCh37:
+        return None
+    rg37 = hl.get_reference(ReferenceGenome.GRCh37.value)
+    rg38 = hl.get_reference(ReferenceGenome.GRCh38.value)
+    if not rg38.has_liftover(rg37):
+        rg38.add_liftover(liftover_ref_path, rg37)
+    return hl.liftover(ht.locus, ReferenceGenome.GRCh37.value)
+
+
+def xpos(ht: hl.Table, **_: Any) -> hl.Expression:
+    return expression_helpers.get_expr_for_xpos(ht.locus)
+
+
+def variant_id(ht: hl.Table, **_: Any) -> hl.Expression:
+    return expression_helpers.get_expr_for_variant_id(ht)
+
+
+def sorted_transcript_consequences(ht: hl.Table, **_: Any) -> hl.Expression:
+    result = hl.sorted(
+        ht.vep.transcript_consequences.map(
+            lambda c: c.select(
+                *SELECTED_ANNOTATIONS,
+                consequence_term_ids=(
+                    c.consequence_terms.filter(
+                        lambda t: ~OMIT_CONSEQUENCE_TERMS.contains(t),
+                    ).map(lambda t: CONSEQUENCE_TERMS_LOOKUP[t])
+                ),
+            ),
+        ).filter(lambda c: c.consequence_term_ids.size() > 0),
+        lambda c: (
+            hl.bind(
+                lambda is_coding, is_most_severe, is_canonical: (
+                    hl.cond(
+                        is_coding,
+                        hl.cond(
+                            is_most_severe,
+                            hl.cond(is_canonical, 1, 2),
```

Comment:
I feel like using constants for these magic numbers would be more readable/maintainable